client.go

```diff
@@ -40,35 +40,69 @@ func init() {
 
 // StatsdClient is a client library to send events to StatsD
 type StatsdClient struct {
-	conn             net.Conn
-	addr             string
-	prefix           string
-	eventStringTpl   string
-	Logger           Logger
-	conn_type        string
-	reconnect_ticker *time.Ticker
+	conn            net.Conn
+	addr            string
+	prefix          string
+	eventStringTpl  string
+	Logger          Logger
+	connType        string
+	reconnectTicker *time.Ticker
 }
 
-// NewStatsdClient - Factory
-func NewStatsdClient(addr string, prefix string) *StatsdClient {
+// ConfigurationFunc is a typedef for a function that configures some aspect of
+// a StatsdClient instance
+type ConfigurationFunc func(*StatsdClient)
+
+// AutoReconnect returns a ConfigurationFunc that causes the StatsdClient to automatically
+// recreate its underlying connection on the specified interval.
+func AutoReconnect(interval time.Duration) ConfigurationFunc {
+	return func(client *StatsdClient) {
+		if interval > 0 {
+			client.reconnectTicker = time.NewTicker(interval)
+		}
+	}
+}
+
+// UseLogger returns a ConfigurationFunc that makes the StatsdClient use the specified Logger
+// implementation, rather than the default implementation.
+func UseLogger(logger Logger) ConfigurationFunc {
```

Comment:
Small nitpick, but why not use the `With` prefix naming convention that we typically use when we employ this pattern?

Reply:
I didn't know that we had really solidified that as a convention.  I know we use it in some places.

If you look at the 3 links that I sent to @Jonnymcc earlier to show our usage of the functional options pattern, you'll see that they all use different naming conventions:

https://github.com/HBOCodeLabs/apiclients/blob/master/catalog/catalog.go#L36-L67
https://github.com/HBOCodeLabs/hurley-kit/blob/master/tracing/middleware.go#L62-L114
https://github.com/HBOCodeLabs/hurley-kit/blob/master/metrics/metrics.go#L130-L156

I don't have strong opinions on the func names in this particular PR though, so I'm good with changing them if you do.

Reply:
@yxiehbo @jlutzhbo Just changed them to use the "With" prefix.  PTAL.